pkg/bpmn_engine/engine_jobs_activated.go

```diff
@@ -18,16 +19,32 @@ type activatedJob struct {
 	processDefinitionKey     int64
 	elementId                string
 	createdAt                time.Time
+	scope                    variable_scope.VarScope
+	localScope               variable_scope.VarScope
 }
 
 // ActivatedJob represents an abstraction for the activated job
 // don't forget to call Fail or Complete when your task worker job is complete or not.
 type ActivatedJob interface {
-	ProcessInstance
+	// GetInstanceKey get instance key from processInfo
+	GetInstanceKey() int64
+	// GetCreatedAt  get job create time
+	GetCreatedAt() time.Time
+	// GetState get instance state
+	GetState() process_instance.State
 
 	// GetKey the key, a unique identifier for the job
 	GetKey() int64
 
+	// GetVariable the varaible from variable scope  include local scope
+	GetVariable(key string) interface{}
+
+	// SetVariable set variable into variable scope
+	SetVariable(key string, value interface{})
```

Comment:
Setting variables just on the activatedJob interface is not good.
I consider we do support User Tasks and Service Tasks.
I would like to alter the process instance variables ONLY when a task is completed OR failed,
but never "in-between".
Please adjust, so that variables are set, when ```.Fail(vars)``` or ```.Complete(vars)``` is called.

Reply:
> Setting variables just on the activatedJob interface is not good. I consider we do support User Tasks and Service Tasks. I would like to alter the process instance variables ONLY when a task is completed OR failed, but never "in-between". Please adjust, so that variables are set, when `.Fail(vars)` or `.Complete(vars)` is called.

What about leaving this part of job to issue 55 which was opened to trace it ?